### ISSUE TITLE
Datasource ElasticSearch Index field not filled in Grafana version 9.4.7

### ISSUE THREAD

**Issue by GPegel** [grafana/helm-charts#2306]
Hi there, I was trying to provision my Grafana datasources setup in ArgoCD while using a piece of yaml like this:

```  logging_datasources.yaml:
    apiVersion: 1
    datasources:
      - name: Logging HomeLab
        type: elasticsearch
        url: https://logging.homelab.local
        access: proxy
        database: "[logs-]YYYY.MM.DD"
        isDefault: false
        jsonData:
          interval: Daily
          timeField: "@timestamp"
          esVersion: 7.10.0
```

When I deploy this configuration while using Grafana version 9.4.7 I see that the datasource is being created and all fields are being added except the "database" field. As you can see, the "Index name" field is not being filled correctly.

<img width="760" alt="Scherm­afbeelding 2023-03-29 om 10 22 56" src="https://user-images.githubusercontent.com/10278496/228477264-81c4f834-dee4-47d6-b391-35be99fbef32.png">

After the deployment I've checked with Postman to see if all the fields are being filled but also in Postman the "database" field is empty.

<img width="710" alt="Scherm­afbeelding 2023-03-29 om 10 20 10" src="https://user-images.githubusercontent.com/10278496/228471971-11370ee6-37ad-4f0a-97eb-f88dea4cef00.png">

To see if I did something wrong I've downgraded Grafana from 9.4.7 to version 9.2.3 and then a new deploy of the same configuration as shown above, all fields including the "database" field is being filled.

<img width="751" alt="Scherm­afbeelding 2023-03-29 om 10 25 08" src="https://user-images.githubusercontent.com/10278496/228473247-84ea66af-7bc5-4639-82e3-8e9f9c094bc3.png">

Do I need another way to add those "database" field in the most recent version of Grafana?

**Comment by plalwa** [comment]
Similar issue, but with loki and tempo and mimir, datasource provisioning from helm-charts is not working.

``` 
  datasources:
    datasources.yaml:
      datasources:
      - name: Mimir
        type: prometheus
        uid: mimir
        access: proxy
        orgId: 1
        url: http://metrics-mimir-gateway/prometheus
        basicAuth: false
        isDefault: true
        version: 1
        editable: true
        jsonData:
          httpHeaderName1: 'X-Scope-OrgID'
        secureJsonData:
          httpHeaderValue1: 'metamonitoring'
          ```

**Comment by GPegel** [comment]
It's solved in version 10.1.4
